src/midi/fluid_midi.c

```diff
@@ -1687,7 +1687,8 @@ new_fluid_player(fluid_synth_t *synth)
     fluid_player_set_playback_callback(player, fluid_synth_handle_midi_event, synth);
     fluid_player_set_tick_callback(player, NULL, NULL);
     player->use_system_timer = fluid_settings_str_equal(synth->settings,
-                               "player.timing-source", "system");
+                               "player.timing-source", "system") || synth->idle_timeout;
```

Comment:
Obviously using a sample-timer and an idle-timeout doesn't work. I would vote for issuing a hard failure in this case, explaining that player.timing-source should be set to system explicitly. Rather than silently falling back to system timer. The same behaviour should be applied to the seqeuencer.

Reply:
Yes, good idea. I've added checks and hard failures to both player and sequencer in 9be14b9.

Reply:
Hard failure is good for API users, but I think the command-line program should be a little more helpful. If a user has switched idle-timeout on but hasn't explicitly chosen sample timers on the command line, then it should switch to system timers automatically but issue a warning.

Reply:
Yes, the fluidsynth executable can be user-friendly and magically adjust it. I don't mind.